File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    // 서로 다른 옷의 조합
    // 얼굴 개수 * 상의 개수 * 하의 개수 * 겉옷 개수 + 모든 옷의 개수
    
    const clothMap = clothes.reduce((acc, [clothName, clothType]) => {
        if (clothType in acc) {
            return {
                ...acc, 
                [clothType]: acc[clothType] + 1, 
            };
        } else {
            return {
                ...acc, 
                [clothType]: 1,
            }
        }
    }, {})
    
    const total = clothes.length
    const values = Object.values(clothMap);
    const count = values.reduce((acc, cur) => acc * (cur + 1), 1);
    
    if (values.length === 1) {
        return total;
    }
    
    return count - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    from collections import defaultdict
    
    # 각 종류별 의상 개수를 센다.
    category_counts = defaultdict(int)
    for name, category in clothes:
        category_counts[category] += 1
    
    # 각 종류별 (개수 + 1)을 곱해준다.
    result = 1
    for count in category_counts.values():
        result *= (count + 1)
    
    # 아무것도 선택하지 않은 경우를 제외한다.
    return result - 1